src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstodeliusupdate/resource/SyncResource.kt

```diff
@@ -0,0 +1,51 @@
+package uk.gov.justice.digital.hmpps.hmppsregisterstodeliusupdate.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppsregisterstodeliusupdate.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.hmppsregisterstodeliusupdate.services.CourtRegisterSyncService
+import uk.gov.justice.digital.hmpps.hmppsregisterstodeliusupdate.services.SyncStatistics
+
+@RestController
+@Validated
+@RequestMapping(name = "Sync To Delius", path = ["/sync"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class SyncResource(
+  private val courtRegisterSyncService: CourtRegisterSyncService
+) {
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_REF_DATA') and hasAuthority('SCOPE_write')")
+  @Operation(
+    summary = "Sync all court details from HMPPS Court Register",
+    description = "Updates court information, role required is MAINTAIN_REF_DATA",
+    security = [SecurityRequirement(name = "MAINTAIN_REF_DATA", scopes = ["write"])],
```

Comment:
Ooh I like this, seems like the right place to mention the role and scope.  Should the role be `ROLE_MAINTAIN_REF_DATA` though as that's what is added to the client credentials?

Reply:
I think Spring actually allows for the missing ROLE at the start (but I do usually add this in as I prefer it as it's more explicit - I will update it).  Also looks like some missing tests on that endpoint.